### ISSUE TITLE
Unrecognized status from pull for 'Already exists' or 'Status: Image is up to date for'

### ISSUE THREAD

**Issue by sza-1** [ansible/ansible#13302]
It looks like Ansible does not work with some good statuses.

Target OS: Fedora 22
docker 1.8.2
python-docker-py 1.1.0
Ansible 1.9.4

Task:

```
- name: run jenkins container
  become: yes
  docker:
    image: jenkins
    restart_policy: always
    state: reloaded
```

Result:

```
failed: [10.2.184.250] => {"changes": ["{\"stream\":\"Trying to pull repository docker.io/library/jenkins ... \"}\r\n", "{\"status\":\"Pulling from library/jenkins\",\"id\":\"latest\"}\r\n", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"5679b9b90e09\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"ea6bab360f56\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"0f062bc85662\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"1db7de304924\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"c8aab46f49f7\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"181e8ba79abb\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"d74c94329c2b\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"bae2b1c607c7\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"b30cb5f1d088\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"6796b7682b3e\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"d80ddc7f43a1\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"6705ebcea7a3\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"1074a4f2c00e\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"09e6bb48b2ec\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"09333b8f77d4\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"0047e1bdf597\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"8ae6c2d005e1\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"3c10b1b411c9\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"a7bfbb6ef5f0\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"f724efdb13d5\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"b1c0539faed7\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"655f8d581e7d\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"68987b2eccc9\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"68abbe28fd9a\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"bb92d9ed5d86\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"3438bbe4145a\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"489aa0c461e6\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"59971592a1ab\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"e3535a452202\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"ed029a846932\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"c716e7aa26f1\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"5bf7480d46c0\"}", "{\"status\":\"Already exists\",\"progressDetail\":{},\"id\":\"ee572ce7ce25\"}", "{\"status\":\"Digest: sha256:e33a1565ecb5075f984cf48ba7f0aec321aa67115421012790cd797081dbe7ce\"}\r\n", "{\"status\":\"Status: Image is up to date for docker.io/jenkins:latest\"}\r\n", "{}\r\n"], "failed": true, "status": ""}
msg: Unrecognized status from pull.

FATAL: all hosts have already failed -- aborting
```

**Comment by sza-1** [comment]
Workaround:

```
# XXX: workaround for Ansible bug, run before docker task
- name: pull jenkins image
  become: yes
  command: docker pull jenkins
```

**Comment by abadger** [comment]
Thanks for reporting a problem!  Looks like this was fixed here: https://github.com/ansible/ansible-modules-core/pull/2268
backported to stable-1.9 branch here: b5419a026796c56df32dbd72569690c60321885b

So this should be fixed in either ansible-2.0.0 or ansible-1.9.5 when they are released.

Note: since this is a bug on a module from the core repo, it should have been filed against: https://github.com/ansible/ansible-modules-core/  Just something to keep in mind for next time :-)

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
